core/trino-main/src/test/java/io/trino/operator/BenchmarkGroupByHashOnTpch.java

```diff
@@ -0,0 +1,486 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slices;
+import io.trino.spi.Page;
+import io.trino.spi.PageBuilder;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.DictionaryBlock;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.DoubleType;
+import io.trino.spi.type.IntegerType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeOperators;
+import io.trino.spi.type.VarcharType;
+import io.trino.sql.gen.JoinCompiler;
+import io.trino.type.BlockTypeOperators;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.RunnerException;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static io.trino.jmh.Benchmarks.benchmark;
+import static io.trino.operator.UpdateMemory.NOOP;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+/**
+ * This class attempts to emulate aggregations done while running tpch queries on unpartitioned data.
+ * The data has been acquired on a single node Trino using sf10 scale.
+ * This is by no means a proper simulation of those queries, but merely a less biased version of BenchmarkGroupByHash
+ */
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(3)
+@Warmup(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 20, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkGroupByHashOnTpch
```

Comment:
While I think these benchmarks are beneficial, I think we might be overoptimizing for TPCH use cases. Could we just parametrize various use cases instead of trying to match TPCH data characteristic? Maybe we could use TPCH datagen directly instead of simulating already artificial data?

Reply:
I would either parametrize it and not pretend to be related to TPCH or use actual TPCH data.

BTW: take a look at `io.trino.benchmark.GroupByAggregationSqlBenchmark`. I don't think its used, but maybe it's worth reviving. For one, I think that framework uses generated TPCH instead of pre-generated data

Reply:
> Maybe we could use TPCH datagen directly

It takes about 10 minutes to generate sf10. Numbers for smaller scale factors are not correlated enough with the bigger ones

Reply:
Actually the purpose of this PR is to optimize bigint aggregation, bot add some fancy benchmarks so maybe I should just delete the commit? WDYT?